packages/data-point/lib/reducer-path/factory.js

```diff
@@ -13,27 +13,103 @@ const REDUCER_PATH = (module.exports.type = 'ReducerPath')
 function ReducerPath () {
   this.type = REDUCER_PATH
   this.name = ''
+  this.body = undefined
   this.asCollection = false
 }
 
 module.exports.ReducerPath = ReducerPath
 
+/**
+ * @param {*} source
+ * @returns {boolean}
+ */
 function isPath (source) {
   return _.isString(source) && source.charAt(0) === '$'
 }
 
 module.exports.isPath = isPath
 
+/**
+ * @param {Accumulator} acc
+ * @returns {*}
+ */
+function getAccumulatorValue (acc) {
+  return acc.value
+}
+
+module.exports.getAccumulatorValue = getAccumulatorValue
+
+/**
+ * @param {string} jsonPath
+ * @param {Accumulator} acc
+ * @returns {*}
+ */
+function getFromAccumulator (jsonPath, acc) {
+  return _.get(acc, jsonPath)
+}
+
+module.exports.getFromAccumulator = getFromAccumulator
+
+/**
+ * @param {string} jsonPath
+ * @param {Accumulator} acc
+ * @returns {*}
+ */
+function getFromAccumulatorValue (jsonPath, acc) {
+  return _.get(acc.value, jsonPath)
+}
+
+module.exports.getFromAccumulatorValue = getFromAccumulatorValue
+
+/**
+ * @param {string} jsonPath
+ * @param {Accumulator} acc
+ * @returns {*}
+ */
+function mapFromAccumulatorValue (jsonPath, acc) {
+  if (Array.isArray(acc.value)) {
+    return _.map(acc.value, jsonPath)
+  }
+
+  return null
+}
+
+module.exports.mapFromAccumulatorValue = mapFromAccumulatorValue
+
+/**
+ * @param {string} jsonPath
+ * @param {boolean} asCollection
+ * @returns {Function}
+ */
+function getPathReducerFunction (jsonPath, asCollection) {
+  if (jsonPath === '.' || _.isEmpty(jsonPath)) {
+    return getAccumulatorValue
+  }
```

Comment:
Not to take this PR off topic, but this is where we'd do #52, correct? Something like

```javascript
if (jsonPath === '.') {
  return util.deprecate(getAccumulatorValue, "Accessing the root accumulator value with '$.' will be deprecated in v2 of data-point. Use '$' to access the root value to avoid a breaking change.")
}

if (_.isEmpty(jsonPath)) {
  return getAccumulatorValue
}
```

Reply:
Yep! You should open a PR for that :)

Reply:
Will do. It'll have to wait for this PR to be merged, of course.